src/main/java/dev/su5ed/sinytra/connector/service/ConnectorLoaderService.java

```diff
@@ -29,16 +31,40 @@ public String name() {
 
     @Override
     public void initialize(IEnvironment environment) {
-        Runnable original = FMLLoader.progressWindowTick;
-        FMLLoader.progressWindowTick = () -> {
-            // Hacky way to run invoke prelaunch outside launch plugin iteration,
-            // prevents crash with REIPC. I'm not going to blame them here since
-            // modlauncher provides users with very limited prelaunch hook accessibility
-            ConnectorEarlyLoader.preLaunch();
+        VarHandle provider = uncheck(() -> TRUSTED_LOOKUP.findStaticVarHandle(ImmediateWindowHandler.class, "provider", ImmediateWindowProvider.class));
+        MethodHandle addReads = uncheck(() -> TRUSTED_LOOKUP.findVirtual(Module.class, "implAddReads", MethodType.methodType(void.class, Module.class, boolean.class)));
+
+        ImmediateWindowProvider original = (ImmediateWindowProvider) provider.get();
+        ImmediateWindowProvider newProvider = new ImmediateWindowProvider() {
+
+            @Override
+            public void updateModuleReads(ModuleLayer layer) {
+                var fm = layer.findModule("forge");
+                if (fm.isPresent()) {
+                    try {
+                        addReads.invoke(original.getClass().getModule(), fm.get(), true);
```

Comment:
What happens if we omit the addReads call? I tried running the game without this, and it worked without any errors.
```java
@Override
public void updateModuleReads(ModuleLayer layer) {
    ConnectorEarlyLoader.preLaunch();
    original.updateModuleReads(layer);
}
```

Reply:
This completely skips `preLaunch` in my tests.